core/src/main/java/org/apache/accumulo/core/iterators/SortedKeyValueIterator.java

```diff
@@ -147,4 +147,13 @@
    *              if not supported.
    */
   SortedKeyValueIterator<K,V> deepCopy(IteratorEnvironment env);
+
+  /**
+   * Closes the Iterator. This should be overridden by the implementing class that has access to <tt>SortedKeyValueIterator source</tt> provided in the
```

Comment:
> OrIterator and CountingIterator throw UnsupportedOperationException on init and set the source in the constructor

Ok, I see where you're coming from now. We should definitely *not* treat those as the norm. The fact that they work is an edge condition, not our recommended way of them functioning.